core/src/chain/header.rs

```diff
@@ -0,0 +1,108 @@
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash as _Hash, Hasher};
+
+use chain::entry::Entry;
+use chain::chain::SourceChain;
+
+/// Properties defined in HeadersEntrySchema from golang alpha 1
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[allow(non_snake_case)]
+pub struct Header {
+    /// the type of this entry
+    /// system types may have associated "subconscious" behavior
+    Type: String,
+    /// ISO8601 time stamp
+    Time: String,
+    /// link to the immediately preceding header, None is valid only for genesis
+    HeaderLink: Option<u64>,
+    /// mandatory link to the entry for this header
+    EntryLink: u64,
+    /// link to the most recent header of the same type, None is valid only for the first of type
+    TypeLink: Option<u64>,
+    Signature: String,
```

Comment:
All these fields should be formatted `snake_case`

Reply:
@thedavidmeister I accidentally approved 😅 but I think you know what to do

Reply:
@sphinxc0re this is intentional as per https://github.com/holochain/holochain-proto/blob/4d1b8c8a926e79dfe8deaa7d759f930b66a5314f/entry_headers.go#L7

all the getters are snake case, but i'm trying to keep the struct that is serialized by serde closer to the original for now

i extended the doc comment to explain this

Reply:
If it's only about serialization, then serde is able to rename the fields on serialization.

Reply:
OK then, i'll make the property names match the methods and i put a tx up to configure serde like that - https://github.com/holochain/holochain-rust/issues/75